contracts/BSCValidatorSet.sol

```diff
@@ -707,17 +799,13 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return true;
   }
 
-  function _forceMaintainingValidatorsExit(Validator[] memory _validatorSet) private returns (Validator[] memory unjailedValidatorSet){
+  function _forceMaintainingValidatorsExit(Validator[] memory _validatorSet, bytes[] memory _voteAddrs) private returns (Validator[] memory unjailedValidatorSet, bytes[] memory unjailedVoteAddrs){
     uint256 numOfFelony = 0;
     address validator;
     bool isFelony;
 
     // 1. validators exit maintenance
-    uint256 i;
-    // caution: it must loop from the endIndex to startIndex in currentValidatorSet
-    // because the validators order in currentValidatorSet may be changed by _felony(validator)
-    for (uint index = currentValidatorSet.length; index > 0; index--) {
-      i = index - 1;  // the actual index
+    for (uint i = 0; i < currentValidatorSet.length; i++) {
```

Comment:
Any reason to change this? since  `currentValidatorSet may be changed by _felony(validator)`

Reply:
Fixed